### PR TITLE
Ensure valid target when targeting control plane

### PR DESCRIPTION
**Pull request by grolu**
Ensure valid target is returned by the target builder

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```feature user

```

### CODE REVIEW

File: pkg/target/target_builder.go [hunk 1]
```diff
@@ -164,49 +164,58 @@ func (b *targetBuilderImpl) SetShoot(ctx context.Context, name string) TargetBui
...
+	if err != nil {
+		return err
+	}
+
+	shoot, err := gardenClient.FindShoot(ctx, t.WithShootName(name).AsListOption())
+	if err != nil {
+		return fmt.Errorf("failed to fetch shoot: %w", err)
+	}
+
+	if t.ProjectName() == "" {
```
Comment (holgerkoser) [inline pkg/target/target_builder.go:203]:
```suggestion
	if t.Project == "" {
```

Comment (holgerkoser) [reply pkg/target/target_builder.go:203]: Maybe also for GardenName because we need to use targetImpl here.

**Review by holgerkoser — APPROVED** [review]
Apart from the cosmetic suggestion to not use the getter of the interface but directly the property on the struct, the PR looks very good to me. 
/lgtm

**Review by petersutter — APPROVED** [review]
/lgtm

File: pkg/target/target_builder.go (hunk 1 above)
Comment (grolu) [reply pkg/target/target_builder.go:203]: Yes, I was wondering the same - so let's always use the property I will change it

**Pull request merged**